include/transport/time_queue.h

```diff
@@ -360,11 +387,19 @@ namespace qtransport {
                 auto& [bucket_index, value_index, expiry_tick] = _queue.at(_queue_index);
                 auto& bucket = _buckets.at(bucket_index);
 
-                if (value_index >= bucket.size() || ticks > expiry_tick) {
+                if (value_index >= bucket.size() || (ticks > expiry_tick && !in_spike(ticks))) {
+                    std::cerr << "===> front Object has expired"
+                              << " queue_index: " <<_queue_index << " queue_size: " << _queue.size()
+                              << " value_index: " << value_index
+                              << " bucket_index: " << bucket_index
+                              << " bucket_size: " << bucket.size()
+                              << " tick_delta: " << _timer_ctx.delta
+                              << " " << ticks << " > " << expiry_tick
```

Comment:
Don't love printing in the container, but I suppose we need some info right now? This could get pretty spammy though.

Reply:
This is a debug only. Problem is time queue doesn't have a logger right now and we are still debugging drops. This should really be removed, but I left it here for now so that we can see it happening.